xds/src/main/java/com/linecorp/centraldogma/xds/internal/CentralDogmaSnapshotResources.java

```diff
@@ -27,55 +26,62 @@
 import com.google.common.hash.Hashing;
 import com.google.protobuf.Message;
 
-import com.linecorp.centraldogma.common.Revision;
+import com.linecorp.armeria.common.annotation.Nullable;
 
 import io.envoyproxy.controlplane.cache.ResourceVersionResolver;
 import io.envoyproxy.controlplane.cache.Resources;
+import io.envoyproxy.controlplane.cache.Resources.ResourceType;
 import io.envoyproxy.controlplane.cache.SnapshotResources;
 import io.envoyproxy.controlplane.cache.VersionedResource;
 
 final class CentralDogmaSnapshotResources<T extends Message> extends SnapshotResources<T> {
 
     public static <T extends Message> SnapshotResources<T> create(
-            Iterable<T> resources, Revision revision) {
+            Map<String, Map<String, VersionedResource<T>>> resources, ResourceType resourceType) {
         final ImmutableMap.Builder<String, VersionedResource<T>> versionedResourcesMap = ImmutableMap.builder();
         final ImmutableMap.Builder<String, T> resourcesMap = ImmutableMap.builder();
-        for (T resource : resources) {
-            final String resourceName = Resources.getResourceName(resource);
-            versionedResourcesMap.put(resourceName, VersionedResource.create(resource));
-            resourcesMap.put(resourceName, resource);
+        for (Map<String, VersionedResource<T>> value : resources.values()) {
+            value.values().forEach(versionedResource -> {
+                final String resourceName = Resources.getResourceName(versionedResource.resource());
+                versionedResourcesMap.put(resourceName, versionedResource);
+                resourcesMap.put(resourceName, versionedResource.resource());
+            });
         }
-        return new CentralDogmaSnapshotResources<>(versionedResourcesMap.build(), resourcesMap.build(),
-                                                   Integer.toString(revision.major()));
+        return new CentralDogmaSnapshotResources<>(
+                versionedResourcesMap.build(), resourcesMap.build(), resourceType);
```

Comment:
Question) I understood this map contains all resources for all resources under the xds project.
I think this approach is fine since there is probably a low possibility of hash collision - however, I'm wondering if it may be simpler + less resources will be used if a `CentralDogmaXdsResources` is created per repository instead.

I just wanted to understand the reasoning behind the current design before moving forward

Reply:
`CentralDogmaSnapshotResources`, which `CentralDogmaXdsResources` creates, has to have all resources so we can't create multiple `CentralDogmaXdsResources`.
However, when we support node groups later, we will create multiple `CentralDogmaXdsResources`s https://github.com/line/centraldogma/blob/70aa9a65938fdfbb1bb2a9e279873f526047288d/xds/src/main/java/com/linecorp/centraldogma/xds/internal/ControlPlanePlugin.java#L91